### ISSUE TITLE
add `loadstreaming` and `savestreaming` functions

### ISSUE THREAD

**Issue by ssfrr** [JuliaIO/FileIO.jl#77]
IIUC `load` and `save` are generally one-off operations, i.e. `load("somefile.wav")` loads the full wave file into memory.

Sometimes though, you have huge files that you might not want to load into memory all at once. In my `LibSndFile` package I also export `loadstream` and `savestream` functions that instead return special stream types that you can read/write chunks of audio from.

@jongwook's MP3 package also has the same functionality, but it's not clear who should own those functions. It seems like it would make sense to have them in `FileIO` as they use all the same query tooling. In fact, we implement `load` and `save` as [wrappers](https://github.com/JuliaAudio/LibSndFile.jl/blob/8abfe8a5f62669f9d72b05ff1f7f9b91ce6ecfa2/src/LibSndFile.jl#L203-L212) around the streaming versions.

Do these functions seem like they could fit into the FileIO architecture? What are the video packages currently doing for this sort of thing?

**Comment by timholy** [comment]
Why doesn't the existing `Stream` interface cover this?

Demo:
```jl
julia> using FileIO

julia> fn = Pkg.dir("FileIO", "test", "files", "magic1.tiff")
"/home/tim/.julia/v0.5/FileIO/test/files/magic1.tiff"

julia> s = open(query(fn))
FileIO.Stream{FileIO.DataFormat{:TIFF},IOStream}(IOStream(<file /home/tim/.julia/v0.5/FileIO/test/files/magic1.tiff>),Nullable{String}("/home/tim/.julia/v0.5/FileIO/test/files/magic1.tiff"))

julia> position(s)
0

julia> skipmagic(s)
FileIO.Stream{FileIO.DataFormat{:TIFF},IOStream}(IOStream(<file /home/tim/.julia/v0.5/FileIO/test/files/magic1.tiff>),Nullable{String}("/home/tim/.julia/v0.5/FileIO/test/files/magic1.tiff"))

julia> position(s)
4

julia> read(s, 100);

julia> position(s)
104
```

**Comment by timholy** [comment]
(If this was just a documentation issue, we should fix that.)

**Comment by ssfrr** [comment]
this was addressed in #78.
